prow/apis/prowjobs/v1/types.go

```diff
@@ -247,9 +247,9 @@ type ProwJobStatus struct {
 	// ProwJob.
 	JenkinsBuildID string `json:"jenkins_build_id,omitempty"`
 
-	// PrevReportState stores the previous reported prowjob state
+	// PrevReportStates stores the previous reported prowjob state per reporter
 	// So crier won't make duplicated report attempt
-	PrevReportState ProwJobState `json:"prev_report_state, omitempty"`
+	PrevReportStates map[string]ProwJobState `json:"prev_report_states, omitempty"`
```

Comment:
since nothing else should be using this field yet, I think I'm ok to
/lgtm
/hold
the rest is local to crier and generally seems OK